src/CSET/operators/plot.py

```diff
@@ -429,6 +429,131 @@ def _plot_and_save_vertical_line_series(
     plt.close(fig)
 
 
+def _plot_and_save_histogram_series(
+    cube: iris.cube.Cube, filename: str, title: str, vmin: float, vmax: float, **kwargs
+):
+    """Plot and save a histogram series.
+
+    Parameters
+    ----------
+    cube: Cube
+        2 dimensional Cube of the data to plot as histogram.
+        Plotting options are fixed:
+        density=True, histtype='bar',stacked=True to ensure that
+        a probability density is plotted using matplotlib.pyplot.hist
+        to plot the probability density so that the area under
+        the histogram integrates to 1.
+        stacked is set to True so the sum of the histograms is
+        normalized to 1.
+        ax.autoscale is switched off and the ylim range
+        is preset as (0,1) to make figures comparable.
+    filename: str
+        Filename of the plot to write.
+    title: str
+        Plot title.
+    vmin: float
+        minimum for colourbar
+    vmax: float
+        maximum for colourbar
+    """
+    fig = plt.figure(figsize=(8, 8), facecolor="w", edgecolor="k")
+    # reshape cube data into a single array to allow for a single histogram.
+    # Otherwise we plot xdim histograms stacked.
+    cube_data_1d = (cube.data).flatten()
+    plt.hist(cube_data_1d, density=True, histtype="bar", stacked=True)
+    ax = plt.gca()
+    # try seaborn for a box whisker plot:
+    # https://python-graph-gallery.com/34-grouped-boxplot/
+
+    # Add some labels and tweak the style.
+    ax.set(
+        title=title,
+        xlabel=f"{cube.name()} / {cube.units}",
+        ylabel="normalised probability density",
+        ylim=(0, 1),
+        xlim=(vmin, vmax),
+    )
+    #    ax.autoscale()
+
+    # Save plot.
+    fig.savefig(filename, bbox_inches="tight", dpi=150)
+    logging.info("Saved line plot to %s", filename)
+    plt.close(fig)
+
+
+def _plot_and_save_postage_stamp_histogram_series(
+    cube: iris.cube.Cube,
+    filename: str,
+    title: str,
```

Comment:
This is a good example of when line plots would be benefiical. I would recommend that the plots are collapsed into one plot rather than do as postage stamps. It would gives a  cleaner view of the ensemble distribution.

Reply:
I added a further plot function that plots histograms from an ensemble in a single plot or a postage stamp plot. With the example data this currently only producing a plot for a single member and not for all members even though I am looping over the stamp_coordinate = "realization". @jfrost-mo Maybe you can take a look please?

Reply:
![image](https://github.com/user-attachments/assets/050bf7bf-fea1-406c-9895-294e3193672b)

![image](https://github.com/user-attachments/assets/a1178a77-9edc-42b4-989d-b6deb148522d)